lib/hasher/include/ADLER32Strategy.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef _ADLER32_STRATEGY_HPP_
+#define _ADLER32_STRATEGY_HPP_
+
+#include "HashStrategy.hpp"
+#include <string>
+#include <openssl/sha.h>
```

Comment:
If `ADLER32Strategy.hpp` doesn't depend on any `<openssl/sha.h>` symbols, then this `#include` can be moved into the cpp file.

Reply:
Done for all hash strategies.